ducktape/tests/test.py

```diff
@@ -47,3 +52,55 @@ def run_standalone(cls):
 
         test.log_start()
         test.run()
+
+
+class TestLoader(Logger):
+
+    def is_test_file(self, file_name):
+        """test_*.py or *_test.py"""
+        pattern = "(^test_.*\.py$)|(^.*_test\.py$)"
+        return re.match(pattern, file_name) is not None
+
+    def discover(self, base_dir):
+        test_classes = []
+
+        for pwd, dirs, files in os.walk(base_dir):
+            for f in files:
+                if not self.is_test_file(f):
+                    continue
+
+                # Try all possible module imports for given file
+                path = os.path.abspath(os.path.join(pwd, f))
+                path_pieces = path[:-3].split("/")
+                while len(path_pieces) > 0:
+                    module_name = '.'.join(path_pieces)
+                    # Try to import the current file as a module
+                    try:
+                        module = importlib.import_module(module_name)
+                        self.logger.info("Successfully imported " + module_name)
+                    except Exception as e:
+                        self.logger.warn("Could not import " + module_name + ": " + e.message)
```

Comment:
This generates a lot of noise at the WARN level. This should probably be at DEBUG or even TRACE since it looks like this is going to attempt to load a bunch of things that aren't modules.